app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt

```diff
@@ -1,22 +1,197 @@
+import org.p2p.wallet.moonpay.model.MoonpaySellTransaction.SellTransactionStatus
+import org.p2p.wallet.newsend.ui.NewSendFragment
+import org.p2p.wallet.sell.ui.lock.SellTransactionDetails
+import org.p2p.wallet.send.model.AddressState
+import org.p2p.wallet.send.model.NetworkType
+import org.p2p.wallet.send.model.SearchResult
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.copyToClipBoard
+import org.p2p.wallet.utils.cutMiddle
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.viewbinding.getColor
+import org.p2p.wallet.utils.viewbinding.getHtmlString
+import org.p2p.wallet.utils.viewbinding.getString
 import org.p2p.wallet.utils.viewbinding.viewBinding
-import org.p2p.wallet.R
+import org.p2p.wallet.utils.withArgs
+import timber.log.Timber
+
+private const val ARG_DETAILS = "ARG_DETAILS"
 
-class SellTransactionDetailsBottomSheet : BaseMvpBottomSheet<
-    SellTransactionDetailsContract.View, SellTransactionDetailsContract.Presenter>(
-    R.layout.dialog_send_transaction_details
-) {
+class SellTransactionDetailsBottomSheet :
+    BaseMvpBottomSheet<SellTransactionDetailsContract.View, SellTransactionDetailsContract.Presenter>(
+        R.layout.dialog_send_transaction_details
+    ),
+    SellTransactionDetailsContract.View {
 
-    override val presenter: SellTransactionDetailsContract.Presenter by inject()
+    companion object {
+        fun show(fm: FragmentManager, details: SellTransactionDetails) {
+            SellTransactionDetailsBottomSheet()
+                .withArgs(ARG_DETAILS to details)
+                .show(fm, SelectTokenBottomSheet::javaClass.name)
+        }
+    }
+
+    override val presenter: SellTransactionDetailsContract.Presenter = SellTransactionDetailsPresenter()
     private val binding: DialogSendTransactionDetailsBinding by viewBinding()
 
+    private val details: SellTransactionDetails by args(ARG_DETAILS)
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_Rounded
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        setupViews()
+        expandToFitAllContent()
+    }
+
+    private fun onActionButtonClicked(action: SellTransactionDetailsButtonAction) {
+        when (action) {
+            SellTransactionDetailsButtonAction.SEND -> {
+                val recipient = SearchResult.AddressOnly(AddressState(details.receiverAddress, NetworkType.SOLANA))
+                replaceFragment(NewSendFragment.create(recipient = recipient))
+            }
+            SellTransactionDetailsButtonAction.CLOSE -> {
+                dismissAllowingStateLoss()
+            }
+            SellTransactionDetailsButtonAction.TRY_AGAIN -> {
+                // todo should be a task somewhere to implement try again feature
+            }
+        }
+    }
+
+    private fun setupViews() = with(binding.layoutDetails) {
```

Comment:
big big method =(

Reply:
i know, right? 
i can extract it to class if it helps..